packages/matter.js/src/protocol/ControllerCommissioningHandler.ts

```diff
@@ -0,0 +1,783 @@
+            breadcrumb: this.lastBreadcrumb,
+        }
+    }
+
+    /**
+     * Step 11-12
+     * 11: If the Commissionee both supports it and requires it, the Commissioner SHALL configure the operational network
+     * at the Commissionee using commands such as AddOrUpdateWiFiNetwork (see Section 11.8.7.3, “AddOrUpdateWiFiNetwork
+     * Command”) and AddOrUpdateThreadNetwork (see Section 11.8.7.4, “AddOrUpdateThreadNetwork Command”).
+     * A Commissionee requires network commissioning if it is not already on the desired operational network.
+     * A Commissionee supports network commissioning if it has any NetworkCommissioning cluster instances.
+     * A Commissioner MAY learn about the networks visible to the Commissionee using ScanNetworks command
+     * (see Section 11.8.7.1, “ScanNetworks Command”).
+     * 12: The Commissioner SHALL trigger the Commissionee to connect to the operational network using ConnectNetwork
+     * command (see Section 11.8.7.9, “ConnectNetwork Command”) unless the Commissionee is already on the desired operational network.
+     */
+    private async configureNetwork() {
+        if (this.commissioningOptions.wifiNetwork === undefined && this.commissioningOptions.threadNetwork === undefined) {
+            logger.debug("Commissioning step 11: No network credentials are configured");
+            return {
+                code: CommissioningStepResultCode.Skipped,
+                breadcrumb: this.lastBreadcrumb,
+            }
+        }
+        if (this.collectedCommissioningData.networkFeatures !== undefined && this.collectedCommissioningData.networkStatus !== undefined) {
+            const rootNetworkFeatures = this.collectedCommissioningData.networkFeatures.find(({ endpointId }) => endpointId === 0)?.value;
+            if (rootNetworkFeatures?.wiFiNetworkInterface) {
+                if (this.commissioningOptions.wifiNetwork !== undefined) {
+                    logger.debug("Commissioning step 11: Configuring WiFi network ...");
+                    return await this.configureNetworkWifi();
+                } else {
+                    logger.debug("Commissioning step 11: WiFi network could be commissioned but is not configured");
+                }
+            }
+            if (rootNetworkFeatures?.threadNetworkInterface) {
+                if (this.commissioningOptions.threadNetwork !== undefined) {
+                    logger.debug("Commissioning step 11: Configuring Thread network ...");
+                    return await this.configureNetworkThread();
+                } else {
+                    logger.debug("Commissioning step 11: Thread network could be commissioned but is not configured");
+                }
+            }
+            if (!rootNetworkFeatures?.ethernetNetworkInterface) {
+                throw new CommissioningError("Commissioning step 11: Did not found any Wifi or Thread network interface to configure on Root Endpoint");
```

Comment:
I think "Commissioning step 11: " is redundant because you prefix in the catch clause around the main loop, right?

Reply:
done